hs_geographic_feature_resource/forms.py

```diff
@@ -0,0 +1,254 @@
+                        Field('northlimit', css_class=field_width),
+                        Field('eastlimit', css_class=field_width),
+                        Field('southlimit', css_class=field_width),
+                        Field('westlimit', css_class=field_width),
+                        )
+
+        super(OriginalCoverageFormHelper, self).__init__(allow_edit, res_short_id, element_id, element_name, layout, element_name_label='Spatial Reference', *args, **kwargs)
+
+
+class OriginalCoverageForm(forms.Form):
+    projection_string = forms.CharField(max_length=2048, required=False, label='Coordinate String', widget=forms.Textarea())
+    projection_name = forms.CharField(max_length=256, required=False, label='Coordinate Reference System')
+    datum = forms.CharField(max_length=256, required=False, label='Datum')
+    unit = forms.CharField(max_length=256, required=False, label='Unit')
+
+    northlimit = forms.FloatField(label='North Extent', widget=forms.TextInput())
+    eastlimit = forms.FloatField(label='East Extent', widget=forms.TextInput())
+    southlimit = forms.FloatField(label='South Extent', widget=forms.TextInput())
+    westlimit = forms.FloatField(label='West Extent', widget=forms.TextInput())
+
+
+    def __init__(self, allow_edit=True, res_short_id=None, element_id=None, *args, **kwargs):
+        super(OriginalCoverageForm, self).__init__(*args, **kwargs)
+        self.helper = OriginalCoverageFormHelper(allow_edit, res_short_id, element_id, element_name='OriginalCoverage')
+        self.delete_modal_form = None
+        self.number = 0
+        self.allow_edit = allow_edit
+        self.errors.clear()
+
+        if not allow_edit:
+            for field in self.fields.values():
+                field.widget.attrs['readonly'] = True
+                field.widget.attrs['style'] = "background-color:white;"
+
+    def clean(self):
+        # modify the form's cleaned_data dictionary
+        super(OriginalCoverageForm, self).clean()
+
+        is_form_errors = False
+        extent={}
+        for bound in ('northlimit', 'eastlimit', 'southlimit', 'westlimit'):
+            if self.cleaned_data.get(bound, None):
+                v=self.cleaned_data[bound]
+                if isinstance(v, float):
+                    extent[bound]=self.cleaned_data[bound]
+                else:
+                    self._errors[bound] = ["Data for %s is not float" % bound]
+                    is_form_errors = True
+                    break
+            else:
+                self._errors[bound] = ["Data for %s is missing" % bound]
+                is_form_errors = True
+                break
+
+        if not is_form_errors:
+            self.cleaned_data["northlimit"] = extent["northlimit"]
+            self.cleaned_data["eastlimit"] = extent["eastlimit"]
+            self.cleaned_data["southlimit"] = extent["southlimit"]
+            self.cleaned_data["westlimit"] = extent["westlimit"]
```

Comment:
Can be simplified with `self.cleaned_data.update(extent)` so that the fields involved are not hard coded in multiple places